src/panels/lovelace/entity-rows/hui-cover-entity-row.ts

```diff
@@ -0,0 +1,84 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+
+import "../components/hui-generic-entity-row.js";
+import "../../../components/ha-cover-controls.js";
+import "../../../components/ha-cover-tilt-controls.js";
+
+import CoverEntity from "../../../util/cover-model.js";
+import { HomeAssistant } from "../../../types.js";
+import { EntityRow, EntityConfig } from "./types.js";
+
+class HuiCoverEntityRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+  private _cover?: CoverEntity;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Entity not configured.");
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    if (!this._cover) {
+      this._cover = new CoverEntity(
```

Comment:
I would prefer if we just extract the isTiltOnly function into a standalone helper and call it directly. That way, if an entity would ever change their support on the fly (they shouldn't really, but we don't have hard rules just yet), we wouldn't break.

Have it as a named export in the cover model file.

```
export const isTiltOnly(stateObj) => …
```

Reply:
Having a big model file was a bad move as it means we always import the whole file, even if we use just 1 function. That's why I want to move to a function approach.